Bio/GenBank/__init__.py

```diff
@@ -62,380 +65,6 @@
-assert not _re_simple_location.match("<1")
-assert not _re_simple_location.match(">99999")
-assert not _re_simple_location.match("join(104..160,320..390,504..579)")
-assert not _re_simple_compound.match("bond(12,63)")
-assert _re_simple_compound.match("join(104..160,320..390,504..579)")
-assert _re_simple_compound.match("order(1..69,1308..1465)")
-assert not _re_simple_compound.match("order(1..69,1308..1465,1524)")
-assert not _re_simple_compound.match("join(<1..442,992..1228,1524..>1983)")
-assert not _re_simple_compound.match("join(<1..181,254..336,422..497,574..>590)")
-assert not _re_simple_compound.match(
-    "join(1475..1577,2841..2986,3074..3193,3314..3481,4126..>4215)"
-)
-assert not _re_simple_compound.match("test(1..69,1308..1465)")
-assert not _re_simple_compound.match("complement(1..69)")
-assert not _re_simple_compound.match("(1..69)")
-assert _re_complex_location.match("(3.9)..10")
-assert _re_complex_location.match("26..(30.33)")
-assert _re_complex_location.match("(13.19)..(20.28)")
-assert _re_complex_location.match("41^42")  # between
-assert _re_complex_location.match("AL121804:41^42")
-assert _re_complex_location.match("AL121804:41..610")
-assert _re_complex_location.match("AL121804.2:41..610")
-assert _re_complex_location.match(
-    "AL358792.24.1.166931:3274..3461"
-)  # lots of dots in external reference
-assert _re_complex_location.match("one-of(3,6)..101")
-assert _re_complex_compound.match(
-    "join(153490..154269,AL121804.2:41..610,AL121804.2:672..1487)"
-)
-assert not _re_simple_compound.match(
-    "join(153490..154269,AL121804.2:41..610,AL121804.2:672..1487)"
-)
-assert _re_complex_compound.match("join(complement(69611..69724),139856..140650)")
-assert _re_complex_compound.match(
-    "join(complement(AL354868.10.1.164018:80837..81016),complement(AL354868.10.1.164018:80539..80835))"
-)
-
-# Trans-spliced example from NC_016406, note underscore in reference name:
-assert _re_complex_location.match("NC_016402.1:6618..6676")
-assert _re_complex_location.match("181647..181905")
-assert _re_complex_compound.match(
-    "join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"
-)
-assert not _re_complex_location.match(
-    "join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"
-)
-assert not _re_simple_compound.match(
-    "join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"
-)
-assert not _re_complex_location.match(
-    "join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"
-)
-assert not _re_simple_location.match(
-    "join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"
-)
-
-_solo_bond = re.compile(r"bond\(%s\)" % _solo_location)
-assert _solo_bond.match("bond(196)")
-assert _solo_bond.search("bond(196)")
-assert _solo_bond.search("join(bond(284),bond(305),bond(309),bond(305))")
```

Comment:
Discussion on https://github.com/biopython/biopython/issues/3600 suggests removing the asserts (which started as regex documentation).

We could move them into the test file (as unittests), but that leaves the question of how best to documentation the regexes?

Reply:
I think keeping the `assert` statements here help with code readability. How to handle this case is something to be considered as part of #3600  . I have left a comment there.